Source/CBLView+REST.m

```diff
@@ -16,16 +16,18 @@ @implementation CBLView (REST)
 
 
 - (CBLStatus) compileFromDesignDoc {
-    if (self.registeredMapBlock != nil)
```

Comment:
This same check you deleted here is still in `View.mapBlock`, which calls this method. That means that querying a view from the native API still has the bug. (We have at least one developer doing this, posting questions on the forum.)
I don't know what the best way to fix that is. Always checking for a design doc when querying any view will affect performance. Maybe add a flag to CBLView that remembers that the map block comes from a design doc, and only re-check the design doc if this flag is set?